server.js

```diff
@@ -36,745 +31,720 @@ inherits(Server, EventEmitter)
-        var isIPv4 = function (peer) { return peer.ipv4 }
-        var isIPv6 = function (peer) { return peer.ipv6 }
-
-        var stats = {
-          torrents: infoHashes.length,
-          activeTorrents: activeTorrents,
-          peersAll: Object.keys(allPeers).length,
-          peersSeederOnly: countPeers(isSeederOnly),
-          peersLeecherOnly: countPeers(isLeecherOnly),
-          peersSeederAndLeecher: countPeers(isSeederAndLeecher),
-          peersIPv4: countPeers(isIPv4),
-          peersIPv6: countPeers(isIPv6),
-          clients: groupByClient()
-        }
+          const isSeederOnly = peer => { return peer.seeder && peer.leecher === false }
+          const isLeecherOnly = peer => { return peer.leecher && peer.seeder === false }
+          const isSeederAndLeecher = peer => { return peer.seeder && peer.leecher }
+          const isIPv4 = peer => { return peer.ipv4 }
+          const isIPv6 = peer => { return peer.ipv6 }
+
+          const stats = {
+            torrents: infoHashes.length,
+            activeTorrents,
+            peersAll: Object.keys(allPeers).length,
+            peersSeederOnly: countPeers(isSeederOnly),
+            peersLeecherOnly: countPeers(isLeecherOnly),
+            peersSeederAndLeecher: countPeers(isSeederAndLeecher),
+            peersIPv4: countPeers(isIPv4),
+            peersIPv6: countPeers(isIPv6),
+            clients: groupByClient()
+          }
 
-        if (req.url === '/stats.json' || req.headers['accept'] === 'application/json') {
-          res.write(JSON.stringify(stats))
-          res.end()
-        } else if (req.url === '/stats') {
-          res.end('<h1>' + stats.torrents + ' torrents (' + stats.activeTorrents + ' active)</h1>\n' +
-            '<h2>Connected Peers: ' + stats.peersAll + '</h2>\n' +
-            '<h3>Peers Seeding Only: ' + stats.peersSeederOnly + '</h3>\n' +
-            '<h3>Peers Leeching Only: ' + stats.peersLeecherOnly + '</h3>\n' +
-            '<h3>Peers Seeding & Leeching: ' + stats.peersSeederAndLeecher + '</h3>\n' +
-            '<h3>IPv4 Peers: ' + stats.peersIPv4 + '</h3>\n' +
-            '<h3>IPv6 Peers: ' + stats.peersIPv6 + '</h3>\n' +
-            '<h3>Clients:</h3>\n' +
-            printClients(stats.clients)
-          )
+          if (req.url === '/stats.json' || req.headers['accept'] === 'application/json') {
+            res.write(JSON.stringify(stats))
+            res.end()
+          } else if (req.url === '/stats') {
+            res.end(`<h1>${stats.torrents} torrents (${stats.activeTorrents} active)</h1>\n<h2>Connected Peers: ${stats.peersAll}</h2>\n<h3>Peers Seeding Only: ${stats.peersSeederOnly}</h3>\n<h3>Peers Leeching Only: ${stats.peersLeecherOnly}</h3>\n<h3>Peers Seeding & Leeching: ${stats.peersSeederAndLeecher}</h3>\n<h3>IPv4 Peers: ${stats.peersIPv4}</h3>\n<h3>IPv6 Peers: ${stats.peersIPv6}</h3>\n<h3>Clients:</h3>\n${printClients(stats.clients)}`
```

Comment:
Much better!

Reply:
I think we can remove the `\n`, as header tags do break lines.

Reply:
solved